content/assignments/hello-world/_index.md

```diff
@@ -0,0 +1,151 @@
+   
+   ``` console 
+   &&&
+   System.String
+   ```
+1. You are ready to run the auto-grading tests.
+
+## Running the Auto-grading Tests
+
+The tests provide feedback about your code before you submit your work.  Use the tests to check your work as you complete the assignment.
+
+The tests are unit tests for you to use to check your work.  We call them auto-graded tests instead of unit tests in these projects. 
+
+Let’s learn how to run the tests in Visual Studio.
+
+1. Each assignment will contain a test **project**.  This project contains all of the auto-grading tests and may have more than one test class.  In the Solution Explorer, navigate to HelloTests.cs inside the `Hello.Tests` project. 
+
+   {{% notice blue "Note" "rocket" %}}
+   You will not need to open the `Using.cs` file.  This file connects the testing project to the testing library.  This is not a test class.
+   {{% /notice %}}
+
+   A **test** is a single method in a test file with `[TestMethod]` above it. 
+
+      ```csharp
+      [TestClass]
+      public class AutogradingTests
+      {
+         [TestMethod]
+         public void TestToDemonstrateWhatATestMethodLooksLike()
+         {
+            //test code here
+         }
+      }
+      ```
+
+      A test class may contain many tests. 
+
+
+      ```csharp
+      [TestClass]
+      public class AutogradingTests
+      {
+         [TestMethod]
+         public void TestToDemonstrateWhatATestMethodLooksLike()
+         {
+            //test code here
+         }
+
+         [TestMethod]
+         public void SecondTestToDemonstrateWhatATestMethodLooksLike()
+         {
+            //test code here
+         }
+      }
+   ```
+1. Run all of your tests using the Test Explorer window.
+   1. Mac Users: 
+      1. **View** > **Tests**  
+      1. Select the green triangle on the left of the Test Explorer. 
+      1. Step through the solution until you find the test that failed. 
+      1. [Guide](https://learn.microsoft.com/en-us/visualstudio/mac/testing?view=vsmac-2022#running-tests) to running tests in Visual Studio for Mac
+   1. Windows Users 
+      1. **Test** > **Windows** > **Test Explorer**
+      1. Select the `Run All` icon to run all of your tests. 
+      1. In the Test Explorer, step through your solution until you find the failed test.
+      1. [Guide](https://learn.microsoft.com/en-us/visualstudio/test/run-unit-tests-with-test-explorer?view=vs-2022) to running tests in Visual Studio.
+
+
+1. Visual Studio opens a Test Pane to display test results every time it runs a test. 
+You should see that one of the tests has failed and the other passed.  We have not touched any code yet. Use the Test Pane to navigate into which specific test has failed.
+
+   In this case, the `TestIfSayHelloReturnsCorrectValue()` test, expected our code to output `Hello, World!` Instead, it provided `&&&` causing it to fail.  This particular test has a message about why it failed.  `&&&` is not the correct message.
```

Comment:
The message is -> Assert.AreEqual failed. Expected:<Hello, World!>. Actual:<&&&>. Incorrect message displayed.

Might be a change to share some debugging tips about unit tests.  Also set the ground work for creating helpful messages for why your test failed.

Reply:
added recommended reading for the debugger.  it's a lot, but if anyone wants to explore it now they have a pathway to do so.  they learn more about it in the unit testing chapter.

Reply:
It took me a second to find this, I feel like it belongs here, thoughts?

Reply:
I'm going to leave the debugger information in the third section. Debugging is more for general troubleshooting and not unit tests.  This information is more for students who want to learn about it themselves.  Later in the early MVC chapters, we will review the debugger in more depth.